### PR TITLE
[FIX/MAINT] fix #2497 and #4140 n_jobs in cov

### LINKED ISSUE #2497: Bug mne.cov.compute_covariance + multiprocessing
Hi,
I just want to mention that I used the mne.cov.compute_covariance function and it works with n_jobs =1 (as in the example) but if I put more jobs, I have the following error message:

File "/usr/lib/python2.7/multiprocessing/pool.py", line 558, in get
raise self._value

PicklingError: Can't pickle <class 'mne.cov._ShrunkCovariance'>: it's not found as mne.cov._ShrunkCovariance

### PR DESCRIPTION
**Pull request by dengemann**
Here I replace the nested class definitions with proper subclassing and delegation. Once Travis is happy I'll update whatsnew.

@larsoner @agramfort

### CODE REVIEW

**Comment by larsoner** [comment]
PEP8

https://travis-ci.org/mne-tools/mne-python/jobs/282226691#L3077

**Comment by dengemann** [comment]
> The code changes are just unindenting right? If so +1 for merge

It's a bit more than that but functionality is not changed + I added some missing tests.

**Comment by dengemann** [comment]
> PEP8

Thanks! Just saw it + attempted to fix it in last commit.

File: mne/cov.py [hunk 1]
```diff
@@ -1071,117 +1071,117 @@ def _auto_low_rank_model(data, mode, n_jobs, method_params, cv,
...
+
+class _RegCovariance(BaseEstimator):
+    """Aux class."""
+
+    def __init__(self, info, grad=0.01, mag=0.01, eeg=0.0,
+                 store_precision=False, assume_centered=False):
+        from sklearn.covariance import EmpiricalCovariance
+        self.estimator = EmpiricalCovariance(
+            store_precision=store_precision,
+            assume_centered=assume_centered)
```
Comment (agramfort) [inline mne/cov.py]: to be fully sklearn compatible self.estimator should be self.estimator_ and should be created in the fit.

**Comment by agramfort** [comment]
besides LGTM

thx @dengemann

**Comment by dengemann** [comment]
At least it does not prevent pickling and those are private classes. I can
address your point though.

**Comment by agramfort** [comment]
let's do this tomorrow. it should not take long

**Comment by dengemann** [comment]
@agramfort have a look before brushing your teeth.

**Comment by agramfort** [comment]
thx @dengemann  !

**Pull request merged**